site/public/js/rainbow-customization.js

```diff
@@ -471,18 +475,98 @@ function showLogButton(responseData) {
+        const formData = new FormData();
         // eslint-disable-next-line no-undef
+        formData.append('csrf_token', csrfToken);
+        formData.append('selected_value', selected_value);
+
+        $.ajax({
+            url: url,
+            type: 'POST',
+            data: formData,
+            processData: false,
+            contentType: false,
+            dataType: 'json',
+            success: function (data) {
+                console.log(data);
+                if (data['status'] === 'success') {
+                    resolve(data);
+                }
+                else {
+                    reject(data['message']);
+                }
+            },
+            error: function (jqXHR, textStatus, errorThrown) {
+                console.log('AJAX error:', jqXHR, textStatus, errorThrown);
+                let errorMsg = `An error occurred: Server response: ${jqXHR.status} ${jqXHR.statusText}`;
+                try {
+                    // Attempt to parse JSON, if there's HTML, this will fail
+                    const responseText = jqXHR.responseText;
+                    const jsonStartIndex = responseText.indexOf('{');
+                    if (jsonStartIndex !== -1) {
+                        const jsonResponse = JSON.parse(responseText.substring(jsonStartIndex));
+                        errorMsg = `${jsonResponse.message || jsonResponse.status}`;
+                    }
+                }
+                catch (e) {
+                    console.error('Failed to parse JSON response', e);
+                }
+                reject(errorMsg);
+            },
+        });
+    });
+}
+
+// eslint-disable-next-line no-unused-vars
+function runBuild() {
+    // eslint-disable-next-line no-undef
+    const url = buildCourseUrl(['reports', 'build_form']);
+
+    sendSelectedValue()
+        .then(() => {
+            $.ajax({
+                type: 'POST',
+                url: url,
+                data: { csrf_token: csrfToken },
+                dataType: 'json',
+                success: function (response) {
+                    console.log(response);
+                    if (response.status === 'success') {
+                        $('#save_status').html('Generating rainbow grades, please wait...');
+                        checkBuildStatus();
+                    }
+                    else {
+                        $('#save_status').html('An error occurred while building');
+                    }
+                },
+                error: function (jqXHR, textStatus, errorThrown) {
+                    console.log('AJAX error:', jqXHR, textStatus, errorThrown);
+                    $('#save_status').html('An error occurred while making the request');
+                },
+            });
+        })
+        .catch((error) => {
+            console.error('Caught error:', error);
+            $('#save_status').html(`An error occurred: ${error}`);
```

Comment:
Using `.html()` opens the door to XSS attacks and should be avoided.  Use `.text()` or similar instead.